src/cf.data.pre

```diff
@@ -181,6 +181,13 @@ DOC_START
 	This option is not yet supported by Squid-3.
 DOC_END
 
+# Options removed in 6.x
+NAME: request_entities
+TYPE: obsolete
+DOC_START
+	Remove this line. HTTP/1.1 defines per-message control over GET/HEAD message body using headers.
```

Comment:
Line way too long for cf.data.pre.

Reply:
we don't have a specific formal limit for cf.data.pre, and this line is only displayed in cache.log

Reply:
> we don't have a specific formal limit for cf.data.pre

The above assertion is hardly a valid excuse for adding excessively long lines. Many bad ideas are not formally prohibited.

> this line is only displayed in cache.log

It is also "displayed" in cf.data.pre which developers have to view and edit, often mimicking nearby informal formatting style. The fewer "can't see the whole line of text" inconveniences and "oh, do not look here, it is not a good example to mimic" exceptions we create for developers, the better.

Does Squid code (that logs obsolete directive documentation to cache.log) logs multiline documentation blobs? If not, then _that_ problem could be a valid excuse for using excessively long documentation lines.

FTR, while there are excessively long obsolete directive documentation lines already (e.g., `maximum_single_addr_tries`), there are also multiline obsolete directive documentation blobs (e.g., `header_access`).

Reply:
> > Over many years the directive has had less and less effect on traffic.
> 
> Do you mean that the default setting of this directive rejected fewer and fewer legitimate messages? How do we know that?

I am estimating from the repeating cycles of people coming to the IETF with specifications to upgrade or replace GET with something that has a body.

Plus the traffic is migrating to HTTP/2 and later which use implicit chunking in the form of DATA frames instead of Content-Length. That seems to be translating into an increase of chunked encoding when they fallback to 1.1. This whole section of code is not run when chunked is used.


> 
> > With the correct use of Content-Length in HTTP/1.1 it does nothing.
> 
> What do you mean? AFAICT, the directive has request rejection effect regardless of whether Content-Length is used correctly (whatever "correct usage" means).

As currently coded in master you are right. However that is counter to a SHOULD in the 1.1 spec - Content-Length with valid value means a body does exist. Once we fix Squid to obey that SHOULD there is no need for admin to turn this on.

Reply:
Resolved in59a3bd1